dimod/include/dimod/constrained_quadratic_model.h

```diff
@@ -51,6 +51,12 @@ class ConstrainedQuadraticModel {
     index_type add_constraint(const abc::QuadraticModelBase<B, I>& lhs, Sense sense, bias_type rhs,
                               const std::vector<T>& mapping);
 
+    /// Add a constraint.
+    /// @param constraint As created by ConstrainedQuadraticModel::new_constraint()
+    /// @exception Throws std::logic_error If the constraint's parent is not this model.
```

Comment:
Only constraints listed in the CQM are actually "counted". This just allows syntax like https://github.com/dwavesystems/dimod/pull/1263/files#diff-37dd454d2b561c6833663a2e5f0f1ddafaf3b698e419e3776ffc28c759c432acR143-R173 where we want to create a constraint, with any potential errors, _before_ adding it to the model. Otherwise an error raised part-way through would result in a partially created model.